### ISSUE TITLE
Saved Models, tf.train.Saver

### ISSUE THREAD

**Issue by IronChariot** [nivwusquorum/tensorflow-deepq#10]
Firstly, let me just say that I love this project, the code is so easy to read and understand, and it blends two things I really wanted to experiment with!

I'm getting to the stage where I want to save/restore model variables after training. I notice that you have a saved_model folder with a .ckpt file for the karpathy game, but I do not see any way to load it using the current notebook.

I've read up on tf.train.Saver, and have tried to use its save and restore functions with partial success (my q_network's weights seem to get restored fine, but the target_q_network's weights do not). Do you have a version of the karpathy_game notebook that was used to create/restore your saved model? If not, could you please advise on what I should keep in mind when setting up tf.train.Saver() and using the save/restore methods?

(EDIT: As is pretty typical for me, I asked this prematurely - I figured out my mistake after a little more fiddling with the code. I had initialised the Saver before the controller was set up, so of course it couldn't save all the variables used by the controller... Sorry!)

**Comment by siemanko** [comment]
Glad to hear that you figured it out! What is your project about?

**Comment by IronChariot** [comment]
For now I've just been making my own simulation (a simplified kind of 'asteroids' game, built off of the karpathy game simulation), which has been pretty fun. The plan is to get some practice tweaking neural network hyper-parameters to allow agents to learn to deal with more and more complex environments. 

Eventually I also want to modify the deep Q controller in other ways in order to understand it better and see what effects certain changes have on the rate of learning and on how complex the agent's strategies can get. I'm still pretty inexperienced when it comes to neural networks (and this is my first project experimenting with tensorflow), so mostly I'm just trying to get my head around how tensorflow works and how DQN is implemented. I'm also watching the continuous branch with interest, will probably be experimenting with that once I'm satisfied with my discrete deep Q experiments.

**Comment by siemanko** [comment]
Sounds awesome, do you have a demo/video I could run? I am curious to see what your simulation looks like. 

btw. be aware that continuous branch is not yet functional! ;-)

**Comment by IronChariot** [comment]
My fork of the project has a functional notebook for the asteroids game (albeit with some bugs I need to sort out before I can say that simulation's finished). I wanted to have a directional agent (like in Karpathy's javascript DQN example) and no walls, so the camera is always focused on the hero (I've also got an option to have a fixed camera, but the hero's sight lines can't see across a wall to the other side right now). Here's a gif: http://imgur.com/4NsgRyQ

Yeah, I remember you mentioning in another issue that the continuous branch wasn't functional yet. It'd be really cool to have an implementation of that though, I'd love to help if I can (once I'm more familiar with tensorflow and I've experimented a bunch with the discrete version...)
